pkg/i2gw/providers/ingressnginx/resource_reader.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingressnginx
+
+import (
+	"context"
+
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw"
+)
+
+// converter implements the i2gw.CustomResourceReader interface.
+type resourceReader struct {
+	conf *i2gw.ProviderConf
+}
+
+// newResourceReader returns a resourceReader instance.
+func newResourceReader(conf *i2gw.ProviderConf) *resourceReader {
+	return &resourceReader{
+		conf: conf,
+	}
+}
+
+func (r *resourceReader) ReadResourcesFromCluster(ctx context.Context, customResources interface{}) error {
+	// ingress-nginx does not have any CRDs.
+	return nil
+}
+
+func (r *resourceReader) ReadResourcesFromFiles(ctx context.Context, customResources interface{}, filename string) error {
+	// ingress-nginx does not have any CRDs.
+	return nil
+}
```

Comment:
It may be easier to save this part of the interface for when we're actually adding support for an implementation that has additional custom resources. Defining it without a clear use case/implementation means that we're probably going to miss some things anyway.

Reply:
I tend to agree with @LiorLieberman on this one. I do think it will be more convenient to have it as a reference point for further implementations, but if you have strong feelings about it I will remove it @robscott